### PR TITLE
Backfill community_emoji data with a DUS 🌱

### PR DESCRIPTION
**Pull request by juliannatetreault**
## What type of PR is this? (check all applicable)

- [x] Preparation for removal ✂️ 
- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents
- Related Issue #15877 

## QA Instructions, Screenshots, Recordings
To QA this PR, please run the data update script (data_updates:run) and ensure that your community name updates accordingly, appending your community emoji to your community name. If your community does not have a community emoji, then the script should run without breaking or appending anything. 

### UI accessibility concerns?
N/A

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I've updated the [Developer Docs](https://developers.forem.com) or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
- [ ] This PR changes the Forem platform and our documentation needs to be
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
- [ ] I've updated the README or added inline documentation
- [ ] I've added an entry to
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
- [x] I will share this change in a [Changelog](https://forem.dev/t/changelog)
      or in a [forem.dev](http://forem.dev) post
- [x] I will share this change internally with the appropriate teams
- [ ] I'm not sure how best to communicate this change and need help
- [ ] This change does not need to be communicated, and this is why not: _please
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?
This DUS should run on all Forems.

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by aitchiss** [comment]
Hey @juliannatetreault 👋  Just doing a wee check on older PRs and wondering if this one is still being worked on?

**Comment by juliannatetreault** [comment]
> Hey @juliannatetreault 👋 Just doing a wee check on older PRs and wondering if this one is still being worked on?

Hey, @aitchiss! Yes, it is. It has been on hold, as I haven't had another cool-down period since opening it where I've had time to pick it back up, but hopefully I'll get to wrapping it up soon!

**Comment by aitchiss** [comment]
@Ridhwana what do you think we should do with this one in light of Julianna's departure (💔)? Shall I assign it to one of us in the admin pod to look at at some point?

**Comment by Ridhwana** [comment]
> @Ridhwana what do you think we should do with this one in light of Julianna's departure (💔)? Shall I assign it to one of us in the admin pod to look at at some point?

@aitchiss regarding this and [this comment](https://github.com/forem/forem/pull/16902#issuecomment-1190390155) I was meant take it to completion during some of my downtime + write a changelog post for this and figure out a communication plan. I haven't had a chance to look into it :( but I yeah we can assign it to one of us in the AE team.

**Comment by aitchiss** [comment]
I'll add it to the project board just so the PR is captured somewhere, and from the comments here then OSS rotation folks should know what's up 🙂

File: spec/lib/data_update_scripts/backfill_community_emoji_spec.rb
Comment (Ridhwana) [inline spec/lib/data_update_scripts/backfill_community_emoji_spec.rb:18]: I tried doing this:
```
    allow(Settings::Community).to receive(:community_name).and_return("Emoji Community")
    allow(Settings::Community).to receive(:community_emoji).and_return("🥐")
```

but the expectation fails when stubbing it out in this way. Whilst I think the approach taken above is okay (but maybe not optimal) I've seen it set like this in other DUS files, I'm curious to know if anyone knows why its failing?

Comment (fdocr) [reply spec/lib/data_update_scripts/backfill_community_emoji_spec.rb:18]: It seems to me like the `allow` stub is used to return the original value `"Emoji Community"` string even after the DUS executes. So this happens:

1. Both `community_name` and `community_emoji` are stubbed with `allow` to return the original values
2. The DUS runs, and assigns the new value
3. The `expect` assertion fails because `Settings::Community.community_name` will continue to return the stubbed value despite the DUS ran

I looked up options and there's a block syntax for the `allow` stubbing but it doesn't work if we used `and_return(...)`, so I'm not sure if there's a way to achieve this. Maybe we just have to deal with assigning them manually in these specs. I actually think it would be best if we do that for all these specs (not just this one where the name should change), to ensure the `expect` clauses are testing the actual result instead of the original stubbed value.

File: lib/data_update_scripts/20220815102733_backfill_community_emoji.rb
Comment (fdocr) [inline lib/data_update_scripts/20220815102733_backfill_community_emoji.rb:6]: **Question:** Should we consider the scenario of the DUS executing on a Forem instance that already has the emoji in their name? I'm not sure if this is even necessary (if it's never meant to happen), but a quick guard clause here might make this DUS slightly less error prone if something like this were to happen

```ruby
return if Settings::Community.community_name.end_with(Settings::Community.community_emoji)
```

Comment (Ridhwana) [reply lib/data_update_scripts/20220815102733_backfill_community_emoji.rb:6]: I don't think it will hurt to do something like that, however we're also following up with a changelog post to let Forem owners know about the change and allow them to remove the community emoji form the name if they would like to. 

I was going to initially use `ends_with?` but then I decided to go with `include?` so that it will pick up if the emoji is anywhere in the string. If the emoji is in the community_name already, then I doubt they'll want us to add another one at the end. I added a test as well.

File: spec/lib/data_update_scripts/backfill_community_emoji_spec.rb
Comment (Ridhwana) [reply spec/lib/data_update_scripts/backfill_community_emoji_spec.rb:18]: Cool thanks Fernando, I'm going to leave it as is for now then if nobody else has any other suggestions.

**Review by fdocr — APPROVED** [review]
LGTM!

**Review by joshpuetz — APPROVED** [review]

**Comment by Ridhwana** [comment]
Re-requesting Reviews as I made some additions to this PR:

Once I run the DUS, it will add the `community_emoji` to the `community_name` and so when we display the `community_name` and `community_emoji` together like in the page title `#{page_title} - #{community_name} #{community_emoji}` it will display the emoji twice. Hence in this PR, I have removed any UI related displays of community_emoji. Once I merge this, I'll also be merging https://github.com/forem/forem/pull/16902 to remove it from the admin and other places too.

**Review by joshpuetz — APPROVED** [review]

**Review by maestromac — APPROVED** [review]
LGTM!

**Pull request merged**
